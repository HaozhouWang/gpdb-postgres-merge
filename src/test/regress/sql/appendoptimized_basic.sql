create table ao_basic_t1 (a int, b varchar) using appendoptimized distributed by (a);

-- Validate that the appendoptimized table access method will be used
-- for this table.
select amhandler from pg_class c, pg_am a where c.relname = 'ao_basic_t1' and c.relam = a.oid;
select count(*) = 1 from pg_appendonly where relid = 'ao_basic_t1'::regclass;

insert into ao_basic_t1 values (1, 'abc'), (2, 'pqr'), (3, 'lmn');

insert into ao_basic_t1 select i, i from generate_series(1,12)i;

select * from ao_basic_t1;

select gp_segment_id, * from ao_basic_t1;

select gp_segment_id, count(*) from ao_basic_t1 group by gp_segment_id;

insert into ao_basic_t1 values (1, repeat('abc', 100000));

select a, length(b) from ao_basic_t1;

create index i_ao_basic_t1 on ao_basic_t1 using btree(a);
select * from ao_basic_t1 where a = 2;

create table heap_t2(a int, b varchar) distributed by (a);
insert into heap_t2 select i, i from generate_series(1, 20)i;

select * from ao_basic_t1 t1 join heap_t2 t2 on t1.a=t2.a where t1.a != 1;


