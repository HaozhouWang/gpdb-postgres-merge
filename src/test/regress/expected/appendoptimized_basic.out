create table ao_basic_t1 (a int, b varchar) using appendoptimized distributed by (a);
insert into ao_basic_t1 values (1, 'abc'), (2, 'pqr'), (3, 'lmn');
insert into ao_basic_t1 select i, i from generate_series(1,12)i;
select * from ao_basic_t1;
 a  |  b  
----+-----
  1 | abc
  1 | 1
 12 | 12
  2 | pqr
  3 | lmn
  2 | 2
  3 | 3
  4 | 4
  7 | 7
  8 | 8
  5 | 5
  6 | 6
  9 | 9
 10 | 10
 11 | 11
(15 rows)

select gp_segment_id, * from ao_basic_t1;
 gp_segment_id | a  |  b  
---------------+----+-----
             2 |  5 | 5
             2 |  6 | 6
             2 |  9 | 9
             2 | 10 | 10
             2 | 11 | 11
             0 |  2 | pqr
             0 |  3 | lmn
             0 |  2 | 2
             0 |  3 | 3
             0 |  4 | 4
             0 |  7 | 7
             0 |  8 | 8
             1 |  1 | abc
             1 |  1 | 1
             1 | 12 | 12
(15 rows)

select gp_segment_id, count(*) from ao_basic_t1 group by gp_segment_id;
 gp_segment_id | count 
---------------+-------
             0 |     7
             1 |     3
             2 |     5
(3 rows)

insert into ao_basic_t1 values (1, repeat('abc', 100000));
select a, length(b) from ao_basic_t1;
 a  | length 
----+--------
  2 |      3
  3 |      3
  2 |      1
  3 |      1
  4 |      1
  7 |      1
  8 |      1
  5 |      1
  6 |      1
  9 |      1
 10 |      2
 11 |      2
  1 |      3
  1 |      1
 12 |      2
  1 | 300000
(16 rows)

create index i_ao_basic_t1 on ao_basic_t1 using btree(a);
select * from ao_basic_t1 where a = 2;
 a |  b  
---+-----
 2 | pqr
 2 | 2
(2 rows)

create table heap_t2(a int, b varchar) distributed by (a);
insert into heap_t2 select i, i from generate_series(1, 20)i;
select * from ao_basic_t1 t1 join heap_t2 t2 on t1.a=t2.a where t1.a != 1;
 a  |  b  | a  | b  
----+-----+----+----
  2 | 2   |  2 | 2
  2 | pqr |  2 | 2
  3 | 3   |  3 | 3
  3 | lmn |  3 | 3
  4 | 4   |  4 | 4
  7 | 7   |  7 | 7
  8 | 8   |  8 | 8
  5 | 5   |  5 | 5
  6 | 6   |  6 | 6
  9 | 9   |  9 | 9
 10 | 10  | 10 | 10
 11 | 11  | 11 | 11
 12 | 12  | 12 | 12
(13 rows)

